src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -152,85 +152,138 @@ public override void InitializeJobExtension(IExecutionContext executionContext,
                 }
-                else
+
+                // Update the checkout task display name if not already set
+                if (string.IsNullOrEmpty(checkoutTask.DisplayName) || string.Equals(checkoutTask.DisplayName, "Checkout", StringComparison.OrdinalIgnoreCase))
                 {
-                    if (checkoutTask.Inputs.ContainsKey(Pipelines.PipelineConstants.CheckoutTaskInputs.Clean))
+                    var repository = RepositoryUtil.GetRepository(executionContext.Repositories, repositoryAlias);
+                    if (repository != null)
                     {
-                        executionContext.Variables.Set(Constants.Variables.Build.RepoClean, checkoutTask.Inputs[Pipelines.PipelineConstants.CheckoutTaskInputs.Clean]);
+                        string repoName = repository.Properties.Get<string>(RepositoryPropertyNames.Name);
+                        string version = RepositoryUtil.TrimStandardBranchPrefix(repository.Properties.Get<string>(RepositoryPropertyNames.Ref));
+                        string path = null;
+                        if (checkoutTask.Inputs.ContainsKey(PipelineConstants.CheckoutTaskInputs.Path))
+                        {
+                            path = checkoutTask.Inputs[PipelineConstants.CheckoutTaskInputs.Path];
+                        }
+                        else
+                        {
+                            path = IOUtil.MakeRelative(repository.Properties.Get<string>(RepositoryPropertyNames.Path), pipelineWorkspaceDirectory);
+                        }
+                        checkoutTask.DisplayName = StringUtil.Loc("CheckoutTaskDisplayNameFormat", repoName, version, path);
                     }
                     else
                     {
-                        executionContext.Variables.Set(Constants.Variables.Build.RepoClean, Boolean.FalseString);
+                        Trace.Info($"Checkout task {checkoutTask.Name} has a repository property {repositoryAlias} that does not match any repository resource.");
                     }
                 }
             }
 
+            // Set variables
+            if (submoduleCheckout.HasValue)
+            {
+                executionContext.Variables.Set(Constants.Variables.Build.RepoGitSubmoduleCheckout, submoduleCheckout.Value.ToString());
+            }
 
-            // Prepare the build directory.
-            executionContext.Output(StringUtil.Loc("PrepareBuildDir"));
-            var directoryManager = HostContext.GetService<IBuildDirectoryManager>();
-            TrackingConfig trackingConfig = directoryManager.PrepareDirectory(
-                executionContext,
-                executionContext.Repositories,
-                workspace);
+            if (repoClean.HasValue)
+            {
+                executionContext.Variables.Set(Constants.Variables.Build.RepoClean, repoClean.Value.ToString());
```

Comment:
can you change this an the one on line 238 to use SetVariable ?

Reply:
Certainly!